### DISCUSSION TITLE
makefile

### DISCUSSION THREAD

**Discussion by saniho** [saniho/apiEnedis#119] [General]
@mdeweerd : comment se passe ton integration
j'ai vu que tu avais fait un merge du /dev,  est-ce que pour toi c'est ok pour livrer une pre-release ?

**Comment by mdeweerd** [comment]
Pour moi le Makefile est ok.  Je te laisse l'exécuter une fois pour appliquer les modifications apportés par les scripts du Makefile (après avoir regardé les autres modifications manuelles).

Le test local passe et en CI aussi.

**Reply by saniho** [reply]
le makefile n'est pas une github action ? il se lance comment ?

**Reply by mdeweerd** [reply]
Désolé, je n'avais pas vu cette question.

Le makefile c'est à lancer en local depuis là ou le fichier se trouve en faisant "make" .

A terme ce serait bien d'avoir cela sous tox comme proposé, mais je ne maitrise pas tox pour le moment et sa mise en place avait échoué.


En CI, les actions étaient déjà sour le workflow github.

**Comment by mdeweerd** [comment]
Je ne peux pas repousser sur les branches même si c'est moi qui est à l'origine de la création.
Donc je passe par un fork.

Pour le moment je soutiens essentiellement sur le flot - j'apporte les évolutions après les avoir mis au point sur un composant que je gères.

**Comment by saniho** [comment]
@mdeweerd, je sais pas si tu as vu, mais j'ai fait un revert sur la merge request suite erreur https://github.com/saniho/apiEnedis/pull/130

je te laisse revoir et faire une new merge

**Reply by mdeweerd** [reply]
Pareil, je ne sais pas si tu as vu, mais j'ai fait des évolutions - le revert sur le revert c'est pas simplé à défaire, mais c'est fait.

Je peux faire un nouveau PR.

**Reply by saniho** [reply]
je vois deux pull request, je peux les checker ou tu as des choses à revoir ?

**Reply by mdeweerd** [reply]
C'est pas pour la même chose.

L'un c'est pour le release, mais il faut que le fichier soit dans le main pour que cela fonctionne.
L'autre pour les vérifs supplémentaires.

Tu peux vérifier.

**Comment by saniho** [comment]
super job, merci, j'ai merge et livré une release...
on va la laisser tourner 1 ou 2 jours et si ok, alors on l'officialisera

**Comment by mdeweerd** [comment]
Très bien, merci - j'ai mis à jour dans mon installation également.

Je viens de voir qu'il y a un répertoire 'myEnedis' qui est créé - j'ajoute #134  et j'en ai profité pour ajouté une icône.

**Reply by saniho** [reply]
j'ai mis un commentaire, tu peux m'expliquer ? car j'ai pas cela de mon coté

**Reply by mdeweerd** [reply]
Sur mon HA, dnas le répertoire custom_components/apiEnedis - il y a les json des appels dedans dans des sous-répertoires qui correspondent au contrats.

```
[core-ssh apiEnedis]$ ls -lrat myEnedis
total 16
drwxr-xr-x    4 root     root          4096 Jan 30 18:35 .
drwxr-xr-x    2 root     root          4096 Jan 31 12:13 215457XXXXX
drwxr-xr-x    2 root     root          4096 Feb 24 00:33 215458XXXXX
drwxr-xr-x    6 root     root          4096 Feb 24 21:41 ..
```

**Reply by mdeweerd** [reply]
Pour info, dans 'custom_components' j'ai un lien symboique vers ce clone git:

```
lrwxrwxrwx    1 root     root            41 Jan 27 21:53 myEnedis -> ../apiEnedis/custom_components/apiEnedis/
```

**Reply by saniho** [reply]
ce n'est pas normal, il devrait se trouver dans /customer_components/archives/myEnedis/xxx

**Reply by mdeweerd** [reply]
J'ai des trucs là aussi (archive sans s) mais cela date de janvier.

Sinon, c'est pas une bonne idée de mettre cela sous custom_components car cela ressemble à un domaine "archive" du coup, à bouger à l'avenir je dirai sous le domaine myEnedis.

**Reply by mdeweerd** [reply]
En fait c'est parce que:
```
        dir_path = os.path.dirname(os.path.realpath(__file__))
        dir_path = dir_path.replace("myEnedis", "archive")
```

le dir_path prend le chemin réel (pas le lien), donc cela ne s'appelle pas "myEnedis", mais apiEnedis.

Je pense qu'il faudra:
- mettre ce répertoire sous 'myEnedis' pas comme '../archive'.
- Modifier donc le hacs.json que je viens de proposer pour mettre "archives" et non "myEnedis" comme j'ai fait.
- Modifier le code pour vérifier si ce répertoire ../archive existe et si oui le bouger (mv) sous myEnedis.
- Renommer apiEnedis en myEnedis (à priori sans impact) + ajuster hacs.json pour cela.

Alternativement, le mettre sous '../config' directement .

**Reply by saniho** [reply]
Sauf que si on met l archive dans myenedis, on  ecrase a chaque version, car hacs remplace le repertoire ... Or le but etait de garder les json de la version davant pour charger par defaut la card...

**Reply by mdeweerd** [reply]
Il y a justement l'option dans hacs.json qui désigne le répertoire à maintenir.
Je l'utilise ailleurs également.

**Reply by saniho** [reply]
j"ai fait le merge et installer la release créé, mais je n'ai pas le resultat escompté, dans config\custom_components\myEnedis\archive, je n'ai que empty.txt

**Comment by mdeweerd** [comment]
J'ai tente de mettre en place un service, mais je me heurte au nom du domaine qui est mixte.

Il faut peut-être se résoudre à rendre le nom du domaine compatible avec les exigence de HA, cela débloquera également d'autres fonctionnalites (HACS Default, icône).

**Reply by saniho** [reply]
En effet, je propose de partir sur myenedis definitivement, mais le truc, c'est que si on change le nom, les utilisateurs vont perdrent leur config actuel...non ?

**Reply by mdeweerd** [reply]
Oui, je pense, donc il faut voir s'il n'y a pas un moyen de récuperer cette config là.  Tout est sous HASS donc peut-être récupérable.
Peut-être qu'il faut préparer le changement : pousser la configuration vers le nouveau nom et mettre à jour ensuite.
Et/ou sauver la config dans un JSON.

Le Pb. c'est que sans config, l'intégration ne démarre pas.

Peut-être pousser la config vers le nouveau nom et renommer le répertoire après.

Enfin, cela devient un peu plus compliqué.

Sinon, maintenir 2 git en attendant: un qu'on mets comme HACS default et l'autre historique à mettre à jour depuis le nouveau tout en maintenant l'ancien nom.

**Reply by saniho** [reply]
penses-tu que coté nom des sensors on peut garder ceux en place ?
car si oui, alors on doit pouvoir juste renommer le repertoire...en faisant attention à garder le nom et la casse des sensors

**Reply by mdeweerd** [reply]
Je crois qu'on peut garder les sensors.

C'est pas juste le répertoire, mais le domaine qui change et c'est lié à la configuration.

**Comment by saniho** [comment]
j'ai fait 2 nouvelles discussion, une pour le probleme de repertoire archive et une pour le "renommage"...celle-ci gardons la pour le makefile

**Reply by mdeweerd** [reply]
ok, j'en avais fait une nouvelle aussi pour un autre sujet.  Je répond essentiellement aux endroits ou les sujets sont ouverts...
